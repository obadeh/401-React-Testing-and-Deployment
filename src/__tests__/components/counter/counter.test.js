import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Counter from '../../../components/counter.js';

describe('<Thing />', () => {
  it('exists at the start of our application', () => {
    let app = shallow(<Counter />);
    expect(app.find('h3').exists()).toBeTruthy();
  });

  it('changes the state on click', ()=> {
    let app = mount(<Counter />);
    let button = app.find('.b1');
    button.simulate('click');
    expect(app.state('count')).toBeTruthy();
    // expect(app.find('span').text()).toContain('false');
  });

  it('renders correctly', () => {
    const tree = renderer.create(<Counter />).toJSON();
    expect(tree).toMatchSnapshot();
  })
});