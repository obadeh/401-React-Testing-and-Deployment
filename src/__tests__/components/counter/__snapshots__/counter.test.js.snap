// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Thing /> renders correctly 1`] = `
<section>
  <h3>
    0
  </h3>
  <h2
    className=""
  >
    
  </h2>
  <button
    className="b1"
    onClick={[Function]}
  >
    <span>
      increment
    </span>
  </button>
  <button
    className="b2"
    onClick={[Function]}
  >
    <span>
      decrement
    </span>
  </button>
</section>
`;
